core/numbers/numbers.py

```diff
@@ -5,12 +5,12 @@
 mod = Module()
 ctx = Context()
 
-digits = "zero one two three four five six seven eight nine".split()
+digit_list = "zero one two three four five six seven eight nine".split()
 teens = "ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen".split()
 tens = "twenty thirty forty fifty sixty seventy eighty ninety".split()
 scales = "hundred thousand million billion trillion quadrillion quintillion sextillion septillion octillion nonillion decillion".split()
 
-digits_map = {n: i for i, n in enumerate(digits)}
+digits_map = {n: i for i, n in enumerate(digit_list)}
 digits_map["oh"] = 0
```

Comment:
Shit, I didn't realize it, but this is a problem. Your new version uses `digits_list` rather than `digits_map` and therefore doesn't include any small numbers that use "oh" for zero! So merging this PR broke backwards compat, my bad.

The data flow through globals in this file is getting kinda hard to follow, we should probably try to confine it to just one place.